frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/AttachmentPortletUtils.java

```diff
@@ -126,7 +127,11 @@ public void serveAttachmentBundle(List<AttachmentContent> attachments, ResourceR
     private void serveAttachmentBundle(List<AttachmentContent> attachments, ResourceRequest request, ResourceResponse response, Optional<String> downloadFileName){
         String filename;
         String contentType;
-        if(attachments.size() == 1){
+        if (!StringUtils.isEmpty(request.getParameter(PortalConstants.ATTACHMENT_NAME))
+                && request.getParameter(PortalConstants.ATTACHMENT_NAME).equalsIgnoreCase("AttachmentBundle.zip")) {
```

Comment:
Thanks for the suggestion @imaykay .
As I see it,  there is no way to change the default zip bundle name and as per the existing code there is a hard coded bundle name given "AttachmentBundle.zip" , if the number of files is greater then one. So if the number of files is greater then 1 then it will always download AttachmentBundle.zip.
Considering these facts ,do we need to add the above checks here? Please suggest. 
If we need to change the approach , is it fine with you to pass only the type in the request attribute .
urlWriter.withParam("attachmentType", "allAttachments");

Reply:
So my understanding of the current code is (of course I might be wrong though ;-) ):

Developers using the tag
- `DisplayDownloadAttachmentBundle`

can give a name for the attachment in the jsp via the
- `name`

attribute, for example:
- `<sw360:DisplayDownloadAttachmentBundle
        name="MyZip.zip"
        attachments="${attachments}"
        contextType="${contextType}"
        contextId="${contextId}" />`

But that name was just ignored until now. And you just added the possibility to really access the name param when the attachment should be downloaded. But in the current manner it is basically a no-op since we only respect it if it matches our default name.

What I was trying to suggest is a way where it will be respected, no matter what the user of the tag wants to have as filename. My personal opinion is that this is how the tag should work. That is why I came up with the algorithm in my first comment.
What I did not regard in the algo is the possibility to pass a filename in the java method params as well. So maybe the algorithm should be:

1. Is there a filename as method param? If yes, use it.
2. Is there a filename as tag attribute? If yes, use it.
3. If both filenames are not given, check bundle size.
3a) If size is 1, use the filename of the single bundle entry
3b) If size is bigger than 1, use our default name "AttachmentBundle.zip"

But I am happy to discuss, maybe on Slack?